src/Plugin/views/field/EntityBundle.php

```diff
@@ -0,0 +1,117 @@
+  /**
+   * Constructs a \Drupal\field\Plugin\views\field\Field object.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Entity\EntityManagerInterface $entity_manager
+   *   The field formatter plugin manager.
+   * @param \Drupal\Core\Field\FormatterPluginManager $formatter_plugin_manager
+   *   The field formatter plugin manager.
+   * @param \Drupal\Core\Field\FieldTypePluginManagerInterface $field_type_plugin_manager
+   *   The field plugin type manager.
+   * @param \Drupal\Core\Language\LanguageManagerInterface $language_manager
+   *   The language manager.
+   * @param \Drupal\Core\Render\RendererInterface $renderer
+   *   The renderer.
+   *  @param \Drupal\Core\Entity\EntityTypeBundleInfoInterface $entityTypeBundleInfo
+   *    The entity type bundle info.
+   *
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityManagerInterface $entity_manager, FormatterPluginManager $formatter_plugin_manager, FieldTypePluginManagerInterface $field_type_plugin_manager, LanguageManagerInterface $language_manager, RendererInterface $renderer, EntityTypeBundleInfoInterface $entityTypeBundleInfo) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition, $entity_manager, $formatter_plugin_manager, $field_type_plugin_manager, $language_manager, $renderer);
+    $this->entityTypeBundleInfo = $entityTypeBundleInfo;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+        $configuration,
+        $plugin_id,
+        $plugin_definition,
+        $container->get('entity.manager'),
+        $container->get('plugin.manager.field.formatter'),
+        $container->get('plugin.manager.field.field_type'),
+        $container->get('language_manager'),
+        $container->get('renderer'),
+        $container->get('entity_type.bundle.info')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function defineOptions() {
+    $options = parent::defineOptions();
+    $options['hide_one_bundle'] = ['default' => TRUE];
+
+    return $options;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildOptionsForm(&$form, FormStateInterface $form_state) {
+    parent::buildOptionsForm($form, $form_state);
+
+    // Option to hide the bundle name if there's only one bundle.
+    $form['hide_one_bundle'] = [
+      '#type' => 'checkbox',
+      '#title' => t('Hide if there\'s only one bundle.'),
+      '#default_value' => $this->options['hide_one_bundle'],
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function renderItems($items) {
+    // Check option setting if we want to display bundle name.
```

Comment:
"// Check option setting if we want to display bundle name." -> this comment is redundant, it didn't tell us anything new. When commenting focus on the "why", not the "what". In this case you can simply omit the comment.